src/api.js

```diff
@@ -251,12 +251,22 @@ var api = {
     var requestFunction = function() {
       return _request(reqOptions, options.logOptions);
     };
+
+    var secureRequest = true;
```

Comment:
This is a fairly large change but IMO safe.  If the options passed to `api.js` include `auth: true` but the resource being accessed is a `http://` endpoint, we don't authorize.